### DISCUSSION TITLE
It should be possible to reference reusable workflows by just their major version, like composite actions

### DISCUSSION THREAD

**Discussion by ghost** [community/community#130777] [Actions] [answered]
### Select Topic Area

Question

### Body

If I want to reuse workflows in an organization it's useful to be able to refer to a version tag. But reusable workflows only allow referring to the exact version: `v1.0.0` and not `v1`.

It's very useful to be able to refer just to the major version (i.e. v1), instead of needing to specify the exact version (v1.0.0). I want the action to automatically use the last minor and patch from the version I chose. Updating this manually is horrible and going to fall by the wayside eventually. On the other hand, I do want to manually update major versions (i.e. v1 to v2).

This is exactly how composite actions work: https://docs.github.com/en/actions/creating-actions/about-custom-actions#good-practices-for-release-management

I think reusable workflows should support this as well. Why? Quoting from the github docs on composite actions:

> If you're developing an action for other people to use, we recommend using release management to control how you distribute updates. Users can expect an action's patch version to include necessary critical fixes and security patches, while still remaining compatible with their existing workflows. You should consider releasing a new major version whenever your changes affect compatibility.
>
> Under this release management approach, users should not be referencing an action's default branch, as it's likely to contain the latest code and consequently might be unstable. Instead, you can recommend that your users specify a major version when using your action, and only direct them to a more specific version if they encounter issues.

**Comment by joshjohanning** [comment, marked as answer]
You can use whatever tag you want when referencing reusable workflows.  Those tags (`v1.0.0` and `v1`) are created manually by action authors. 

So if you are creating the `v1.0.0` tag, you can also create and push the `v1` tag too. 

```sh
git tag v1 --force && git push origin v1 --force
```

I automate this using GitHub actions on new tag creation.

**Reply by ghost** [reply]
Yeah you're right. It's such a common pattern for github actions that I figured that this pattern was handled automatically. But it seems like it isn't. Seems a little awkward to me, but I've dealt with it by adding a major version tag and shifting that on each release with https://www.npmjs.com/package/semantic-release-major-tag.

**Reply by joshjohanning** [reply]
> Yeah you're right. It's such a common pattern for github actions that I figured that this pattern was handled automatically. But it seems like it isn't. Seems a little awkward to me, but I've dealt with it by adding a major version tag and shifting that on each release with https://www.npmjs.com/package/semantic-release-major-tag.

That's a great solution. This action can also help roll the tag. 

https://github.com/marketplace/actions/update-major-version
